app/src/main/java/com/example/triptracker/view/map/RecordScreen.kt

```diff
@@ -643,78 +640,103 @@ fun StartWindow(viewModel: RecordViewModel, context: Context) {
+                          onClick = { toggleRecording(viewModel) },
+                          modifier = Modifier.align(Alignment.CenterVertically).fillMaxHeight(0.6f),
+                          colors =
+                              ButtonDefaults.filledTonalButtonColors(
+                                  containerColor = md_theme_light_onPrimary,
+                                  contentColor = md_theme_light_dark),
+                      ) {
+                        Text(
+                            text = getPauseResumeText(viewModel),
+                            fontSize = 14.sp,
+                            fontFamily = Montserrat,
+                            fontWeight = FontWeight.SemiBold,
+                            color = md_theme_light_dark)
                       }
-                }
+                      Row(
+                          modifier = Modifier.align(Alignment.CenterVertically),
+                          horizontalArrangement = Arrangement.SpaceEvenly) {
+                            Text(
+                                text = "Add spot",
+                                modifier = Modifier.align(Alignment.CenterVertically),
+                                fontSize = 14.sp,
+                                fontFamily = Montserrat,
+                                fontWeight = FontWeight.SemiBold,
+                                color = md_theme_grey)
+                            Spacer(modifier = Modifier.width(20.dp))
+                            IconButton(
+                                onClick = { viewModel.addSpotClicked.value = true },
+                                modifier =
+                                    Modifier.align(Alignment.CenterVertically)
+                                        .size(50.dp)
+                                        .background(md_theme_light_onPrimary, shape = CircleShape),
+                            ) {
+                              Icon(
+                                  imageVector = Icons.Outlined.Add,
+                                  contentDescription = "Add spot",
+                                  tint = md_theme_light_dark,
+                                  modifier = Modifier.size(30.dp))
+                            }
+                          }
+                    }
               }
         }
+      }
+}
+
+fun getPauseResumeText(viewModel: RecordViewModel): String {
+  return if (viewModel.isPaused.value) "Resume" else "Pause"
+}
+
+fun toggleRecordingDescription(viewModel: RecordViewModel) {
+  if (viewModel.isRecording()) {
+    viewModel.stopRecording()
+    viewModel.startDescription()
+  } else {
+    viewModel.startRecording()
+  }
+}
+
+fun getBorderColor(bool: Boolean): Color {
+  return if (isDescriptionEmpty) md_theme_light_error else md_theme_grey
+}
+
+fun toggleRecording(viewModel: RecordViewModel) {
+  if (viewModel.isPaused.value) {
+    viewModel.resumeRecording()
+  } else {
+    viewModel.pauseRecording()
   }
 }
```

Comment:
A little bit sad to have backend like functions inside a file for @Composable functions, but we are forced to do it to lower the complexity measure :/